File: tests-playwright/src/keywords/files.keywords.ts
import { test } from '../fixtures/fixtures'
import type { FilesPage } from '../pages/files/files-page'

export async function action_openFiles(filesPage: FilesPage): Promise<void> {
  await test.step('Open the files page', async () => {
    await filesPage.open()
  })
}

export async function check_filesShellReady(filesPage: FilesPage): Promise<void> {
  await test.step('Verify the upload and download cards render', async () => {
    await filesPage.verifyShellReady()
  })
}

File: tests-selenium/src/test/java/com/coinpulse/selenium/keywords/FilesKeywords.java
package com.coinpulse.selenium.keywords;

import com.coinpulse.selenium.pages.FilesPage;
import io.qameta.allure.Allure;
import org.openqa.selenium.WebDriver;

/**
 * Files keywords: specs call these, never the page object directly.
 */
public class FilesKeywords {

    private final FilesPage filesPage;

    public FilesKeywords(WebDriver driver) {
        this.filesPage = new FilesPage(driver);
    }

    public void actionOpenFiles() {
        Allure.step("Open the files page", filesPage::open);
    }

    public void checkFilesShellReady() {
        Allure.step("Verify the upload and download cards render", filesPage::verifyShellReady);
    }
}
